cmd_exec.c

```diff
@@ -128,3 +121,40 @@ int cmd_exec(context_t *context) {
   free(command);
   return ret;
 }
+
+int cmd_exec_sync(char **command) {
+  pid_t child = fork();
+
+  if (child == 0) { /* child */
+    execvp(command[0], command);
+
+    /* if we get here, there was an error */
+    perror("execvp failed");
+    exit(errno);
+  } else { /* parent */
+    int status = 0;
+    waitpid(child, &status, 0);
+    return WEXITSTATUS(status);
+  }
+}
+
+void cmd_exec_detached(char **command) {
+  pid_t child = fork();
+
+  if (child == 0) { /* child */
+    pid_t grandchild = fork();
+    if (grandchild == 0) { /* grandchild */
```

Comment:
Now that I think of it indeed explicitly ignoring SIGCHLD's (`signal(SIGCHLD, SIG_IGN);`) might also be a good solution - I don't remember if I considered that when I prepared this PR.

Indeed that shows it is useful to better document the rationale behind these approaches :).